src/Phinx/Config/Config.php

```diff
@@ -294,6 +306,58 @@ public function getTemplateClass()
      }
 
     /**
+     * Get the version order.
+     *
+     * @return string
+     */
+    public function getVersionOrder()
+    {
+        return $this->values['version_order'];
+    }
+
+    /**
+     * Set the version order.
+     * @param string $versionOrder The Version Order to set.
+     */
+    public function setVersionOrder($versionOrder = null)
+    {
+        if (!isset($versionOrder)) {
+            $versionOrder = isset ( $this->values['version_order'] ) ? $this->values['version_order'] : null;
+        }
+
+        if (!isset($versionOrder)) {
+            // if the configuration value is missing we use the default version order (creation time)
+            $this->values['version_order'] = self::VERSION_ORDER_CREATION_TIME;
```

Comment:
Oh dear. I led you the wrong way. I forgot that the config class is, essentially, read only, no setters() needed. All guards are in the getters().

And looking at the rest of config, it doesn't really guard against much!

I think the getter can be as simple as:
```php
return $this->values['version_order'] ?: self::VERSION_ORDER_CREATION_TIME;
```